thoth/prescriptions_refresh/handlers/package_downloads.py

```diff
@@ -0,0 +1,215 @@
+    job_config.destination = f"{dataset_id_full}.destination_table"
+
+    query = """
+    SELECT *
+    FROM `bigquery-public-data.pypi.file_downloads`
+    WHERE DATE(timestamp)
+        BETWEEN DATE_SUB(CURRENT_DATE(), INTERVAL 180 DAY)
+        AND CURRENT_DATE()
+    """
+    query_job = client.query(query=query, job_config=job_config)
+
+    rows = query_job.result()
+    packages_downloads_dict = {}
+    for row in rows:
+        packages_downloads_dict[(row.file.project, row.file.version)] = (
+            packages_downloads_dict.get((row.file.project, row.file.version), 0) + 1
+        )
+
+    concatenated_package_version_dict = {
+        package[0] + " " + package[1]: downloads for package, downloads in packages_downloads_dict.items()
+    }
+
+    _LOGGER.info("Creating plots of the number of downloads per package and per package version")
+    _plot_statistics(packages_downloads_dict)
+    _plot_statistics(concatenated_package_version_dict, includes_versions=True)
+
+    packages_total_downloads = _packages_total_downloads(packages_downloads_dict)
+    for project_name in prescriptions.iter_projects():
+        prescription_name = ""
+        for part in map(str.capitalize, project_name.split("-")):
+            prescription_name += part
+        prescription_name += "PackagePopularityWrap"
+
+        downloads_count, popularity_level = _popularity_level(packages_total_downloads, project_name)
+        package_link = os.path.join("https://pypi.org/project/", project_name)
+
+        package_versions_downloads = {
+            (package[0], package[1]): downloads
+            for package, downloads in packages_downloads_dict.items()
+            if package[0] == project_name
+        }
+        popularity_per_version_summary = ""
+        for package_version, downloads in package_versions_downloads:
+            popularity_per_version_summary += f"'{package_version[0]}' version {package_version[1]} has a {_downloads_to_popularity(downloads)} popularity with {downloads} downloads in the last six months. \n"
+
+        prescriptions.create_prescription(
+            project_name=project_name,
+            prescription_name=_PACKAGE_DOWNLOADS_PRESCRIPTION_NAME,
+            content=_PACKAGE_DOWNLOADS_PRESCRIPTION_CONTENT.format(
+                package_name=project_name,
+                prescription_name=prescription_name,
+                popularity_level=popularity_level,
+                downloads_count=downloads_count,
+                package_link=package_link,
+                popularity_per_version_summary=popularity_per_version_summary,
+            ),
+        )
+
+        for package_version, downloads_count in package_versions_downloads.items():
+            prescription_name_per_version = ""
+            for part in map(str.capitalize, package_version[0].split("-")):
+                prescription_name_per_version += part
+            prescription_name_per_version += f"{package_version[1]}PackagePopularityPerVersionWrap"
```

Comment:
Adviser will refuse to register prescriptions that have the same names. Each prescription name has to be unique to have the ability to uniquely reference it.

Reply:
What would be the timestamp in this case?

Reply:
The current date to see when the prescription was refreshed?